### PR TITLE
Add migration stream

### PR DESCRIPTION
**Pull request by jakepruitt**
Transform stream for migration v1 tiles to v2.

cc/ @mapsam @rclark @springmeyer @yhahn

### CODE REVIEW

File: lib/tilelivecopy.js [hunk 1]
```diff
@@ -45,9 +47,11 @@ function tilelivecopy(srcUri, s3url, options, callback) {
...
         if (scheme === 'list') options.listStream = src.createZXYStream();
 
-        if (options.stats) {
-          options.transform = stats;
-        }
+        var transforms = [];
+
+        if (options.stats) transforms.push(stats);
+        if (options.type === 'mbtiles') transforms.push(MigrationStream());
+        if (transforms.length > 0) options.transform = combiner(transforms);
```
Comment (mapsam) [inline lib/tilelivecopy.js]: This looks fancy, mind explaining what's going on here @jakepruitt? From the sounds of `combiner` I imagine this is how we are sending two streams through `tilelive.copy`?

Comment (jakepruitt) [inline lib/tilelivecopy.js]: @mapsam yep! If there's stats, we want the stats stream. If it's an mbtiles we want the migration stream. So we combine them together and throw it into the transform of tilelive-copy.

File: lib/migration-stream.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    var format = tiletype.type(tile.buffer);
+    if (format !== 'pbf') {
+      migrationStream.push(tile);
+      return callback();
+    }
+
+    var info = mapnik.VectorTile.info(tile.buffer);
+
+    if (info.errors) {
+        var err = new Error('Invalid data');
```
Comment (rclark) [inline lib/migration-stream.js]: What kinds of cases are we dealing with here? How can we make this error more informative?

As I read, this occurs if an incoming tile could be set on a vector tile, but not read out of one? Like a V1 tile that cannot for some reason be transformed to V2?

If possible we will want more info in this error, both to convey to the user what actions they can take to remedy the situation, and maybe also so we can debug upstream in mapnik.

File: lib/migration-stream.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        err.code = 'EINVALID';
+        return callback(err);
+    }
+
+    var v2 = info.layers.every(function(layer) {
+        return layer.version === 2;
+    });
+    if (v2 === true) {
```
Comment (rclark) [inline lib/migration-stream.js]: This condition yields no tile. What leads to this scenario, and is this an acceptable outcome?

File: lib/migration-stream.js (hunk 2 above)
Comment (mapsam) [inline lib/migration-stream.js]: I think `info.errors` will report any errors from [this list in mapnik vector tile](https://github.com/mapbox/mapnik-vector-tile/blob/master/src/vector_tile_is_valid.hpp#L52-L103).

Comment (rclark) [inline lib/migration-stream.js]: Let's use those error messages which are far more informative, and IMO not too complex for a user who would potentially see them (uses mbtiles or studio classic). The only thing I might do is `.replace(' message', '')` in the error descriptions you linked.

Comment (jakepruitt) [inline lib/migration-stream.js]: @rclark yeah, the `mapnik.VectorTile.info` method is a lightweight cursory look at the buffer that checks for errors and reports things like layer versions and feature counts. I think it's a great idea to add the extra information to the returned error data. I'll also add it at lines 12-14.

File: lib/migration-stream.js (hunk 3 above)
Comment (jakepruitt) [inline lib/migration-stream.js]: > This condition yields no tile.

The intention here was that if all of the tiles were already v2, we should pass through the buffer without any conversion. Is calling `migrationStream.push(tile)` not enough to pass through the tile buffer?

**Pull request merged**